packages/std/src/testing/mock.rs

```diff
@@ -392,17 +392,110 @@ fn validate_length(bytes: &[u8]) -> StdResult<()> {
-        },
-        transaction: Some(TransactionInfo { index: 3 }),
-        contract: ContractInfo {
-            address: contract_addr,
-        },
+    let mut envs = Envs::new(BECH32_PREFIX);
+    envs.make()
+}
+
+/// A factory type that stores chain information such as bech32 prefix and can make mock `Env`s from there.
+///
+/// It increments height for each mock call and block time by 5 seconds but is otherwise dumb.
+///
+/// In contrast to using `mock_env`, the bech32 prefix must always be specified.
+///
+/// ## Examples
+///
+/// Typical usage
+///
+/// ```
+/// # use cosmwasm_std::Timestamp;
+/// use cosmwasm_std::testing::Envs;
+///
+/// let mut envs = Envs::new("food");
+///
+/// let env = envs.make();
+/// assert_eq!(env.contract.address.as_str(), "food1jpev2csrppg792t22rn8z8uew8h3sjcpglcd0qv9g8gj8ky922ts74yrjj");
+/// assert_eq!(env.block.height, 12_345);
+/// assert_eq!(env.block.time, Timestamp::from_nanos(1_571_797_419_879_305_533));
+///
+/// let env = envs.make();
+/// assert_eq!(env.contract.address.as_str(), "food1jpev2csrppg792t22rn8z8uew8h3sjcpglcd0qv9g8gj8ky922ts74yrjj");
+/// assert_eq!(env.block.height, 12_346);
+/// assert_eq!(env.block.time, Timestamp::from_nanos(1_571_797_424_879_305_533));
+///
+/// let env = envs.make();
+/// assert_eq!(env.contract.address.as_str(), "food1jpev2csrppg792t22rn8z8uew8h3sjcpglcd0qv9g8gj8ky922ts74yrjj");
+/// assert_eq!(env.block.height, 12_347);
+/// assert_eq!(env.block.time, Timestamp::from_nanos(1_571_797_429_879_305_533));
+/// ```
+///
+/// Or use with iterator
+///
+/// ```
+/// # use cosmwasm_std::Timestamp;
+/// use cosmwasm_std::testing::Envs;
+///
+/// let mut envs = Envs::new("food");
+///
+/// for (index, env) in envs.take(100).enumerate() {
+///     assert_eq!(env.contract.address.as_str(), "food1jpev2csrppg792t22rn8z8uew8h3sjcpglcd0qv9g8gj8ky922ts74yrjj");
+///     assert_eq!(env.block.height, 12_345 + index as u64);
+///     assert_eq!(env.block.time, Timestamp::from_nanos(1_571_797_419_879_305_533).plus_seconds((index*5) as u64));
+/// }
+/// ```
+pub struct Envs {
+    contract_address: Addr,
+    last_height: u64,
+    last_time: Timestamp,
+    envs_produced: u64,
+}
+
+impl Envs {
+    pub fn new(
+        bech32_prefix: &'static str, /* static due to MockApi's Copy requirement. No better idea for now. */
+    ) -> Self {
+        let api = MockApi::default().with_prefix(bech32_prefix);
+        Envs {
+            // Default values here for compatibility with old `mock_env` function. They could be changed to anything else if there is a good reason.
+            contract_address: api.addr_make("cosmos2contract"),
+            last_height: 12_344,
+            last_time: Timestamp::from_nanos(1_571_797_419_879_305_533).minus_seconds(5),
+            envs_produced: 0,
+        }
+    }
+
+    pub fn make(&mut self) -> Env {
+        let height = self.last_height + 1;
+        let time = self.last_time.plus_seconds(5);
```

Comment:
Would it make sense to make the time increment configurable?

Reply:
Sure

Reply:
A lot of things are configurable now. See `EnvsOptions`